bagitobjecttransfer/recordtransfer/static/recordtransfer/js/transferform/forms.js

```diff
@@ -530,6 +567,71 @@ $(() => {
     if (typeof MODAL_MODE !== 'undefined' && MODAL_MODE) {
         initializeModalMode();
     }
+
+
+    /**********************************************************************************************
+     * Source information form setup
+     *********************************************************************************************/
+
+    // Context passed from template to JS
+    const sourceInfoContextElement = document.getElementById("py_context_source_info");
+
+    if (sourceInfoContextElement) {
+        const sourceInfoContext = JSON.parse(sourceInfoContextElement.textContent);
+
+        // Toggles Other source type field depending on value of Source type field
+        setupSelectOtherToggle2(
+            sourceInfoContext["id_source_type"],
+            sourceInfoContext["id_other_source_type"],
+            sourceInfoContext["other_type_id"]
+        );
+
+        // Toggles Other source role field depending on value of Source role field
+        setupSelectOtherToggle2(
+            sourceInfoContext["id_source_role"],
+            sourceInfoContext["id_other_source_role"],
+            sourceInfoContext["other_role_id"]
+        );
+
+        // Add faux-required-field to labels
+        document.querySelectorAll(".faux-required-field").forEach((el) => {
+            let label = el.previousElementSibling;
+            if (label && label.tagName === "LABEL") {
+                label.classList.add("faux-required-field");
+            }
+        });
+
+        const sourceTypeSelect = document.getElementById(sourceInfoContext["id_source_type"]);
+        const sourceRoleSelect = document.getElementById(sourceInfoContext["id_source_role"]);
+        const enterManualInfoInputId = sourceInfoContext["id_enter_manual_source_info"];
+        const enterManualInfoInputElement = document.getElementById(enterManualInfoInputId);
+
+        if (enterManualInfoInputElement) {
+            enterManualInfoInputElement.addEventListener("change", function (event) {
+                const selected = this.querySelector("input[type=radio]:checked").value;
+
+                if (selected === "yes") {
+                    document.querySelectorAll(".initially-hidden").forEach((el) => {
```

Comment:
I see why you have two different classes related to hiding elements now: `initially-hidden` is used to mark an element as hideable, and `hidden-item` actually hides it.